### ISSUE TITLE
Default values for configuration variables

### ISSUE THREAD

**Issue by andreiarusoaie** [kframework/k-legacy#754]
There are cases when we want to be able to set some default values for configuration variables. For instance, let us assume the following configuration:

`
configuration <T> <k> $PGM </k> <in stream="stdin"> $IN:List </in> </T>
`

In command line, we pass input to `test.pgm` as usual:

`
krun test.pgm -cIN="ListItem(1) ListItem(2)"
`

Other programs (e.g. `test_1.pgm`) may not require input values. Since there is no way to specify how the initial configuration is built for a specific program,  we have to set $IN for this program too:

`
krun test_1.pgm -cIN=".List"
`

My proposal is to have a 'default' attribute in the configuration for setting the default value of $IN:

`
configuration <T> <k> $PGM </k> <in stream="stdin" default=".List"> $IN:List </in> </T>
`

With this feature, the above command becomes:

`
krun test_1.pgm
`

If there is no default value for $IN in the configuration and -cIN is missing then `krun` should throw an error.

**Comment by radumereuta** [comment]
I don't think that putting the default value as a cell attribute is the best solution. What if I have:
`<a> $A ~> $B </a>` how can I specify the default for both?

**Comment by andreiarusoaie** [comment]
I don't think that having two variables in a single cell is such common, but we can probably have something like:
`<a default$A="" default$B=""> $A ~> $B </a>`

or, maybe we can have configuration files for krun...

**Comment by dlucanu** [comment]
why not

```
<a $A="va" $B="vb"> $A ~> $B </a>
```

**Comment by radumereuta** [comment]
How about `<a> default($A:Sort, 'some term here') ~> default($B:Sort2, 'some other term') </a>`
I don't like the idea of putting the default value for a variable in the list of attributes for a cell. Considering separation of concerns, I don't see the connection between the two. That's why I'm proposing this special term with special semantics at the krun level.

**Comment by dlucanu** [comment]
Any solution that works is fine with me.
